scala/private/rule_impls.bzl

```diff
@@ -548,7 +548,7 @@ def _create_scala_compilation_provider(ctx, ijar, source_jar, deps_providers):
         runtime_deps = runtime_deps,
     )
 
-def _build_deployable(ctx, jars_list):
+def build_deployable(ctx, jars_list):
```

Comment:
if we are going to make this public, we could actually comment it and improve the arguments. One api could be:
```
def merge_jars(single_jar, actions, deploy_jar, main_class, jars):
```
Without taking the full `ctx` we prevent limit what the function can do and make it clearer what the dependencies are.

Then call it with:
```
merge_jars(ctx.executable._singlejar, ctx.actions, ctx.outputs.deploy_jar, getattr(ctx.attr, "main_class", None), jars_list)
```

Reply:
I'll see if I can follow up with a PR for this next.